feat: Update generator layout and improve empty state screens

e-fisher opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

This PR introduces the following changes to the Generator view to improve user experience:

1. Give request list more space, move rule list to the bottom third of the view. The rule list doesn't need much space and most of the time spent will be in request log - with this layout change it's easier to inspect requests and search data required for rules. 

![screencapture 2025-01-22 at 16 35 17](https://github.com/user-attachments/assets/9c27d770-ab6c-475b-b636-0f53ed1f3568)

2. Show empty state messages on newly created recordings. Currently, if you create a new generator and skip the selection of allowed hosts, you'll end up with this view:

![screencapture 2025-01-22 at 16 43 12](https://github.com/user-attachments/assets/e286fe52-2d5c-458d-96a8-afa51f015501)
- Empty states messages are not displayed
- There's guidance on next action user should take

With changes introduced in this PR:
![screencapture 2025-01-22 at 16 45 55](https://github.com/user-attachments/assets/dcc7f461-cbfb-4c85-a9c8-0e8aef236458)

Now the user is guided to 1) select hosts, 2) add a rule

3. Add new Grot illustrations for empty request list and rule state. Previously used 404 illustration gave an impression something was wrong when creating a new generator.


With this being a rather impactful change, feedback and comments are always welcome! 

## Future improvements

Next iteration would be to improve on rule results preview. Editing a rule switches request tab to result preview, which feels counterintuitive to me. We might explore combining request list and preview into one view.

## How to Test

Verify everything still works as expected, try creating generators, editing rules and validating scripts. 

## Checklist

- [x] I have performed a self-review of my code.
- [ ] I have added tests for my changes.
- [ ] I have run linter locally (`npm run lint`) and all checks pass.
- [ ] I have run tests locally (`npm test`) and all tests pass.
- [x] I have commented on my code, particularly in hard-to-understand areas.
<!-- - [ ] Any other relevant item -->

## Screenshots (if appropriate):

## Related PR(s)/Issue(s)

<!-- - <https://github.com/grafana/k6-studio/issues/...> -->

<!-- Does it resolve an issue? -->

<!-- Resolves #ISSUE-ID -->

<!-- Thanks for your contribution! 🙏🏼 -->

going-confetti reviewed (commented):
Looks good for the most part. My biggest concern is that request inspector becomes less usable:
<img width="1390" alt="image" src="https://github.com/user-attachments/assets/cf89cb57-d9af-4874-a70c-b6b959143b92" />
The screenshot above is from my 4k monitor with 2x UI scaling (so, there's "only" 1080px worth of vertical space, but then you need to subtract the heights of the dock and the status bar)

cristianoventura reviewed (commented):
The addition of the empty state messages looks great! 🤩

I noticed that the visualization of the "Content" tab is a bit cramped when compared to before.

<img width="1168" alt="image" src="https://github.com/user-attachments/assets/69f1ec61-bba3-4b5f-b441-9e56cde0bd26" />

I also noticed that the vertical scroll in the Request section doesn't reach the bottom. It happens for all three tabs (Headers, Content and, Cookies)

https://github.com/user-attachments/assets/2d260041-acfc-4ca4-a4b4-2f74a24c7080

Perhaps the "Test rules" section could be contained in the middle of the section and leave the right side for the Request/Response inspector only?

going-confetti commented:
> I also noticed that the vertical scroll in the Request section doesn't reach the bottom. It happens for all three tabs (Headers, Content and, Cookies)

This should be fixed in my PR from yesterday

going-confetti commented:
> Perhaps the "Test rules" section could be contained in the middle of the screen and leave the right side for the Request/Response inspector only?

I think this is a good suggestion, we'd just need to decide where to put the rule editor. Perhaps, that's a good time consider opening it in place of the rule list? 🤔

cristianoventura reviewed (dismissed):
> I think there's room for improvement, but probably better than what we have now 🤔 The automatic switch to rule preview feels intrusive, but that should be solved when/if we merge request list and preview into a single view.

I agree - For now it looks a lot better even on a smaller screen! 🎸

going-confetti reviewed (changes requested):
I updated the branch and wanted to extract the updated EmptyMessage component into its own PR when I noticed that the some of the illustrations aren't consistent (e.g. the stroke width and color of the visor/glasses):
<img width="1000" alt="image" src="https://github.com/user-attachments/assets/01155469-1b5e-4b3e-b8ee-4a48bbeb286b" />

going-confetti commented on src/views/Generator/GeneratorTabs/RequestList.tsx:
```diff
@@ -0,0 +1,107 @@
+  return (
+    <Flex direction="column" height="100%">
+      <ScrollArea scrollbars="vertical">
+        {isRecordingMissing && (
+          <Flex justify="center" align="center" height="100%">
+            <EmptyMessage
+              message="The selected recording is missing, select another one from the top menu"
+              pt="0"
+            />
+          </Flex>
+        )}
+        {!areHostsSelected && (
```
There are at least 5 different cases when we might want to show a message like this, maybe we can address some of them with the same message if it's generic enough: 
1. No recording is selected (if the user creates an empty generator)
2. Selected recording missing
3. Selected recording is empty (an edge case, but still)
4. Allowlist is empty
5. Allowlist is not empty, but the selected hosts are only used for static assets (see example below)

<img width="737" alt="image" src="https://github.com/user-attachments/assets/3be65426-49cc-4021-9d27-39db9f00c432" />

e-fisher commented:
We'll need custom illustrations created for us to match the existing style, I've reverted to default Grot and will add new illustrations once we have them. 

I've also addressed cases you've mentioned:

![screencapture 2025-02-10 at 12 13 28](https://github.com/user-attachments/assets/fbda3770-1086-4ce7-b39e-8a2154d6ac02)

![screencapture 2025-02-10 at 12 13 45](https://github.com/user-attachments/assets/7741b008-3348-4914-9ccd-a64dd5bbdeaf)

![screencapture 2025-02-10 at 12 13 59](https://github.com/user-attachments/assets/543c92f9-5b1d-49c9-a602-96a776603e8a)

![screencapture 2025-02-10 at 12 14 45](https://github.com/user-attachments/assets/40f30e0b-d5ed-432c-8d44-7ad98de0cad9)

The recording related messages now also include the recording dropdown instead of directing the user to the top menu, similarly like hosts related messages, let me know what you think!

going-confetti reviewed (approved):
LGTM!
